test/performance/range/views/view_translate_benchmark.cpp

```diff
@@ -0,0 +1,306 @@
+// ============================================================================
+
+template <typename container_t, typename adaptor_t>
+void random_access_join(benchmark::State & state)
+{
+    container_t c(2000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+    {
+        c[i] = generate_sequence<seqan3::dna4>(6000, 0, 0);
+    }
+
+    std::vector<size_t> access_positions_frame;
+    access_positions_frame.resize(1000);
+    std::mt19937 gen(42);
+    std::uniform_int_distribution<size_t> dis_frame(0, 5);
+
+    for (size_t i = 0; i < 1000; ++i)
+        access_positions_frame[i] = dis_frame(gen);
+
+    std::vector<size_t> access_positions;
+    access_positions.resize(1000);
+    std::uniform_int_distribution<size_t> dis_position(0, 1000 - 1);
+
+    for (size_t i = 0; i < 1000; ++i)
+        access_positions[i] = dis_position(gen);
+
+    auto adaptor = adaptor_t{};
+
+    for (auto _ : state)
+    {
+        container_t s{c};
+        auto v = s | adaptor;
+        for (size_t i = 0; i < c.size(); ++i)
+        {
+            for (size_t j = 0; j < access_positions_frame.size(); ++j)
+            {
+                benchmark::DoNotOptimize(v[(access_positions_frame[j] * 6) + i][access_positions[j]] | views::to_rank);
+            }
+        }
+    }
+}
+
+BENCHMARK_TEMPLATE(random_access_join, std::vector<std::vector<dna4> >, seqan3::detail::translate_join_fn);
+
+// ============================================================================
+//  copy_to_vector translate
+// ============================================================================
+
+template <typename container_t, typename adaptor_t>
+void copy_to_vector(benchmark::State & state)
+{
+    container_t c(2000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+    {
+        c[i] = generate_sequence<seqan3::dna4>(5000, 0, 0);
+    }
+
+    auto adaptor = views::deep{adaptor_t{}};
+
+    for (auto _ : state)
+    {
+        container_t s{c};
+        std::vector<std::vector<std::vector<aa27>>> vec = s | adaptor
+                                                            | views::to<std::vector<std::vector<std::vector<aa27>>>>;
+    }
+}
+
+BENCHMARK_TEMPLATE(copy_to_vector, std::vector<std::vector<dna4> >, seqan3::detail::translate_fn<false>);
+
+// ============================================================================
+//  copy_to_vector translate_join
+// ============================================================================
+
+template <typename container_t, typename adaptor_t>
+void copy_to_vector_join(benchmark::State & state)
+{
+    container_t c(2000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+    {
+        c[i] = generate_sequence<seqan3::dna4>(5000, 0, 0);
+    }
+
+    auto adaptor = adaptor_t{};
+
+    for (auto _ : state)
+    {
+        container_t s{c};
+        auto v = s | adaptor;
+        std::vector<std::vector<aa27>> vec = s | adaptor
+                                               | views::to<std::vector<std::vector<aa27>>>;
```

Comment:
You might need to wrap these in DoNotOptimize as well since the compiler might see that you don't use the variable and optimise all of that out.